src/ShopBundle/objects/db/TShopOrder.class.php

```diff
@@ -501,6 +502,7 @@ public function SendOrderNotification($sSendToMail = null, $sSendToName = null)
 
         TCMSImage::ForceNonSSLURLs(true); // force image urls to non ssl for use in order email
         if (is_null($oMail)) {
+            $this->SetSendOrderNotificationState(false);
```

Comment:
This would mean that in case of a fatal error (email template missing?) sending would be done again?

Reply:
To reset the notification state here make sense to garantee the same behaviour as before - if the mail template does not exist, it is _possible_ that a succeeding process can send the mail successfully.